### ISSUE TITLE
Most samples (e.g., polybench) don't work

### ISSUE THREAD

**Issue by htorst** [spcl/dace#289]
On a clean checkout (782ffd7f30248d44918f8531bad17704a46f51e2):

$ python ./samples/polybench/2mm.py

```Polybench test k2mm, problem size: large
Traceback (most recent call last):
  File "./samples/polybench/2mm.py", line 106, in <module>
    polybench.main(sizes, args, [(3, 'D')], init_array, k2mm)
  File "/raid/htor/projekte/dace/samples/polybench/polybench.py", line 97, in main
    app.run(lambda argv: _main(sizes, args, outputs, init_array, func, argv,
  File "/home/htor/.local/lib/python3.8/site-packages/absl/app.py", line 299, in run
    _run_main(main, args)
  File "/home/htor/.local/lib/python3.8/site-packages/absl/app.py", line 250, in _run_main
    sys.exit(main(argv))
  File "/raid/htor/projekte/dace/samples/polybench/polybench.py", line 97, in <lambda>
    app.run(lambda argv: _main(sizes, args, outputs, init_array, func, argv,
  File "/raid/htor/projekte/dace/samples/polybench/polybench.py", line 63, in _main
    sdfg = func.to_sdfg(*args)
  File "/raid/htor/projekte/dace/dace/frontend/python/parser.py", line 238, in to_sdfg
    return parse_from_function(self, *args, strict=strict)
  File "/raid/htor/projekte/dace/dace/frontend/python/parser.py", line 109, in parse_from_function
    sdfg = function.generate_pdp(*compilation_args)
  File "/raid/htor/projekte/dace/dace/frontend/python/parser.py", line 321, in generate_pdp
    return newast.parse_dace_program(dace_func, argtypes, global_vars,
  File "/raid/htor/projekte/dace/dace/frontend/python/newast.py", line 173, in parse_dace_program
    sdfg, _, _ = pv.parse_program(src_ast.body[0])
  File "/raid/htor/projekte/dace/dace/frontend/python/newast.py", line 1012, in parse_program
    self.visit_TopLevel(stmt)
  File "/raid/htor/projekte/dace/dace/frontend/python/astutils.py", line 292, in visit_TopLevel
    self.visit(node)
  File "/usr/lib/python3.8/ast.py", line 360, in visit
    return visitor(node)
  File "/raid/htor/projekte/dace/dace/frontend/python/newast.py", line 2297, in visit_Assign
    self._visit_assign(node, node.targets[0], None)
  File "/raid/htor/projekte/dace/dace/frontend/python/newast.py", line 2319, in _visit_assign
    results.extend(self._gettype(node.value))
  File "/raid/htor/projekte/dace/dace/frontend/python/newast.py", line 3028, in _gettype
    operands = self.visit(opnode)
  File "/usr/lib/python3.8/ast.py", line 360, in visit
    return visitor(node)
  File "/raid/htor/projekte/dace/dace/frontend/python/newast.py", line 2859, in visit_Call
    result = func(self.sdfg, self.last_state, *args, **keywords)
  File "/raid/htor/projekte/dace/dace/frontend/python/replacements.py", line 41, in _define_local_ex
    name, _ = sdfg.add_temp_transient(shape, dtype, storage=storage)
  File "/raid/htor/projekte/dace/dace/sdfg/sdfg.py", line 1256, in add_temp_transient
    return self.add_array(self.temp_data_name(),
  File "/raid/htor/projekte/dace/dace/sdfg/sdfg.py", line 1131, in add_array
    desc = dt.Array(dtype,
  File "/raid/htor/projekte/dace/dace/properties.py", line 398, in initialize_properties
    init(obj, *args, **kwargs)
  File "/raid/htor/projekte/dace/dace/data.py", line 289, in __init__
    super(Array, self).__init__(dtype, shape, transient, storage, location,
  File "/raid/htor/projekte/dace/dace/properties.py", line 398, in initialize_properties
    init(obj, *args, **kwargs)
  File "/raid/htor/projekte/dace/dace/data.py", line 65, in __init__
    self.dtype = dtype
  File "/raid/htor/projekte/dace/dace/properties.py", line 220, in __set__
    raise TypeError(
TypeError: Invalid type "Constant" for property dtype: expected typeclass
```
something if going wrong with the types but not clear why (the stack depth is somewhat shocking as well :-/). 

As far as I can tall, all polybench examples are affected. Many others work.

**Comment by tbennun** [comment]
Which Python version are you using?

**Comment by htorst** [comment]
Python 3.8.2

**Comment by tbennun** [comment]
Should now be fixed in 5bdaf89a26b36b23cd6881d6b7c0c4f2c333a628. Please verify. Thanks!

**Comment by htorst** [comment]
works, thanks!
